src/charts/axis.js

```diff
@@ -279,23 +293,46 @@ export default function Axis (_container) {
     return Math.ceil(longestLabelApproxWidth / (config.markPanelWidth / labels.length))
   }
 
+  function skipXLAbels () {
+    if (!config.xTickSkip) {
+      return
+    }
+    const labels = cache.xAxisRoot
+      .select(".axis.x")
+      .selectAll(".tick text")
+
+    let xTickSkip = config.xTickSkip
+    if (config.xTickSkip === "auto") {
+      const labelsText = labels.nodes().map(node => node.textContent)
+      xTickSkip = getNumberOfLabelsToSkip(labelsText)
+    }
+    labels.style("display", (d, i) => (i % xTickSkip) ? "none" : "block")
+  }
+
   function drawAxis () {
     cache.xAxisRoot.select(".axis.x")
       .attr("transform", `translate(0, ${config.chartHeight})`)
       .call(cache.xAxis)
 
+    skipXLAbels()
+
     if (config.labelsAreRotated === true || (config.labelsAreRotated === "auto" && cache.xLabelsShouldRotate)) {
       rotateXLabels()
     } else if (config.labelsAreRotated === "auto" && !cache.xLabelsShouldRotate) {
       unRotateXLabels()
     }
 
+    truncateTickText(cache.xAxisRoot, config.maxXLabelCharCount)
```

Comment:
is it necessary to call `truncateTickText` here if it's being called within the `.on("end")` methods for the y-axises?

Reply:
This one calls it specifically for the x axis

Reply:
right!